host/sgx/loadelf.c

```diff
@@ -1244,15 +1271,27 @@ static oe_result_t _sgx_update_enclave_properties(
 {
     oe_result_t result = OE_UNEXPECTED;
 
+    uint64_t elf_oeinfo_pos = 0;
+    uint64_t elf_image_oeinfo_pos = 0;
+    OE_CHECK(image->elf.oeinfo_file_pos > (uint64_t)image->elf.elf.size);
```

Comment:
Just go through and see if the checks you have introduced to compare these variables is necessary. It might be, I just cant figure it out.

Reply:
As part of fuzzing, program headers can be corrupted and forced to increase offset beyond image size. This check catches such issues. Critically looking such elf image fuzzing could be detected if signing infrastructure is working as expected and we don't need above checks. The issue #4056 is a similar kind of issue where fuzzing  forced offsets to move beyond image size and these checks catches them. If we rely on signing tools infrastructure, we can remove such checks. One more catch is that image_size itself could be corrupted as part of fuzzing as discussed in the issue #4058. Hence, I am also not completely confident on adding these checks here. @mingweishih @anakrish @mrragava  can you add your comments and see if these checks are required.

Reply:
@manojrupireddy ELF parsers should limit the trust only to image_size, All the other offsets should treat image_size as base line. Even if the image_size is tampered by fuzzer the offsets calculated should not add-up beyond the allocated (image_size) image region.